### ISSUE TITLE
Variable sized field

### ISSUE THREAD

**Issue by bearsh** [andreax79/python-cstruct#6]
In C the last member of a struct may be of a variable length. It would be nice to have the same feature in cstruct.

**Comment by andreax79** [comment]
Could you please provide an example?

**Comment by bearsh** [comment]
something like this:
```c
struct pkg {
    uint16_t cmd;
    uint16_t length;
    uint8_t data[];
};
```
In C I would allocate enough space for the cmd/length part plus tha actual data. Later I can assign the data. In python I could create the Object, assign cmd/length and also assign some list to data. Once the list is assigned, I may `pack()` it and the `.__size__` attribute gets valid. The other way round, the size is determined during `unpack()` using all remaining data for the data attribute.

I hope this make it clear...

**Comment by shsirk** [comment]
(+1) I had the same thought while testing one of my testcase with it.

**Comment by K4CZP3R** [comment]
(+1)

**Comment by andreax79** [comment]
implemented in version 3.0
